kalite/distributed/static/js/distributed/exercises/views.js

```diff
@@ -807,26 +196,32 @@ window.ExerciseView = Backbone.View.extend({
         var self = this;
 
         var defaults = {
-            seed: Math.floor(Math.random() * 200),
-            framework: "khan-exercises"
+            seed: Math.floor(Math.random() * 200)
         };
 
-        var question_data = $.extend(defaults, question_data);
+        question_data = $.extend(defaults, question_data);
 
         this.data_model.set(question_data);
 
         this.$("#workarea").html("<center>" + gettext("Loading...") + "</center>");
 
+
+        // Khan Exercises now moves the solution area around and fails to put it back again after
+        // So put it back if it is missing
+        if (this.$("#solutionarea").length === 0) {
+            $(".solutionarea-placeholder").after('<div id="solutionarea" class="solutionarea fancy-scrollbar"></div>');
+        }
+
         this.data_model.update_if_needed_then(function() {
 
-            var framework = self.data_model.get("framework");
+            var framework = self.data_model.get_framework();
 
             Exercises.setCurrentFramework(framework);
 
             if (framework == "khan-exercises") {
 
                 if (Khan.loaded === undefined) {
-                    $.getScript(KHAN_EXERCISES_SCRIPT_URL, self.load_exercises_when_ready);
+                    require([KHAN_EXERCISES_SCRIPT_URL], self.load_exercises_when_ready);
```

Comment:
Seems like one of me is lying. Maybe it could just be 'require' further up. $.getScript wasn't playing nicely with the requires we had lying around.